src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_copy_b_emitter.cpp

```diff
@@ -62,34 +79,33 @@ jit_brgemm_copy_b_emitter::jit_brgemm_copy_b_emitter(jit_generator* h, cpu_isa_t
             "compensations subtensor must be {1, m_N_blk}");
     }
 
-    OV_CPU_JIT_EMITTER_ASSERT(!one_of(m_brg_weight_etype, element::bf16, element::i8), "doesn't support precision ", m_brg_weight_etype);
-    const auto repacking_buffer_shape = brgemm_repack->get_repacking_buffer_shape();
-    OV_CPU_JIT_EMITTER_ASSERT(!repacking_buffer_shape.empty(), "Repacking buffer shape mustn't be empty");
-    const auto& LDB = repacking_buffer_shape.back();
-
     const auto& brg_src_etype = brgemm_repack->get_src_element_type();
     m_brg_weight_etype = brgemm_repack->get_input_element_type(0);
     m_brgemmVNNIFactor = brgemm_repack->get_brgemm_vnni_factor();
+    OV_CPU_JIT_EMITTER_ASSERT(one_of(m_brg_weight_etype, element::f32, element::bf16, element::i8), "doesn't support precision ", m_brg_weight_etype);
 
     const auto use_amx = mayiuse(avx512_core_amx) && brg_src_etype != ov::element::f32 &&
                          (m_K_blk % m_brgemmVNNIFactor == 0) && (m_N_blk % m_brgemmVNNIFactor == 0);
 
     const auto src_dt = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::ElementTypeToDataType(brg_src_etype));
     const auto wei_dt = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::ElementTypeToDataType(m_brg_weight_etype));
 
-    init_brgemm_copy(m_kernel, leading_dimension, m_inner_N_block, m_inner_N_tail, LDB, m_K_blk, use_amx, src_dt, wei_dt);
+    // Note: 4D format tags are used just to force the needed OneDNN primitive creation.
+    // However, the generated primitive can be also applied to tensors with other ranks
+    const auto format = wei_stride != 0 ? m_transpose ? dnnl_adbc : dnnl_acbd : m_transpose ? dnnl_abdc : dnnl_abcd;
```

Comment:
Minor: could you please use parentless here or refactor a little bit? It looks difficult to get it for me at least 😄

Reply:
Refactored, please take a look (especially on the comment with justification why these formats are used)